examples/Examples/ConstrData.hs

```diff
@@ -0,0 +1,145 @@
+{-# LANGUAGE UndecidableInstances #-}
+
+module Examples.ConstrData (tests) where
+
+import Data.String (fromString)
+import qualified GHC.Generics as GHC
+import Generics.SOP
+
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (testCase)
+
+import Utils
+
+import Plutarch.Api.V1
+import Plutarch.DataRepr (PDataFields, PIsDataReprInstances (PIsDataReprInstances))
+import Plutarch.Lift (PConstanted, PLifted)
+import Plutarch.Prelude
+import Plutarch.Unsafe (punsafeCoerce)
+
+import Plutus.V1.Ledger.Api
+import qualified PlutusTx
+
+{- |
+  We can defined a data-type using PDataRecord, with labeled fields.
+
+  With an appropriate instance of 'PIsDataRepr', we can automatically
+  derive 'PDataFields'.
+-}
+newtype Triplet (a :: PType) (s :: S)
+  = Triplet
+      ( Term
+          s
+          ( PDataRecord
+              '[ "x" ':= a
+               , "y" ':= a
+               , "z" ':= a
+               ]
+          )
+      )
+  deriving stock (GHC.Generic)
+  deriving anyclass (Generic)
+  deriving anyclass (PIsDataRepr)
+  deriving
+    (PlutusType, PIsData, PDataFields)
+    via (PIsDataReprInstances (Triplet a))
+
+data PVehicle (s :: S)
+  = PFourWheeler (Term s (PDataRecord '["_0" ':= PInteger, "_1" ':= PInteger, "_2" ':= PInteger, "_3" ':= PInteger]))
+  | PTwoWheeler (Term s (PDataRecord '["_0" ':= PInteger, "_1" ':= PInteger]))
+  | PImmovableBox (Term s (PDataRecord '[]))
+  deriving stock (GHC.Generic)
+  deriving anyclass (Generic)
+  deriving anyclass (PIsDataRepr)
```

Comment:
Why have separate lines for the anyclass derives?

Reply:
Yeah we've kinda been following this style throughout the entire repo.

Reply:
It's been on accident, it's not a good style.

Reply:
We probably want to fix that as a cleanup task - do you want it done here in this example for the time being?

Reply:
Yeah let's make sure we don't add more bad code.

Reply:
done!